src/core/core.scale.js

```diff
@@ -302,20 +316,21 @@ var Scale = Element.extend({
 
 		// IMPORTANT: from this point, we consider that `this.ticks` will NEVER change!
 
-		// BACKWARD COMPAT: synchronize `_ticks` with labels (so potentially `this.ticks`)
-		for (i = 0, ilen = labels.length; i < ilen; ++i) {
-			label = labels[i];
-			tick = ticks[i];
-			if (!tick) {
-				ticks.push(tick = {
-					label: label,
+		if (!ticks) {
```

Comment:
Could leave lines above unchanged and check `!ticks.length` here?
Also shouldn't we use `me.ticks.length` rather than `labels.length` as `ilen` in this loop?
And what if those lengths are not equal?

Reply:
It minimizes a bit better to check just `!ticks` and I think is also more understandable that there's not a real value there yet

Agreed on changing loop termination condition